Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: test-capability: CAP_LINUX_IMMUTABLE is not available in unprivileged containers
Origin: upstream, commit:058a07635f3ff70cc99943dcf4f2a079bc9c28b9
Forwarded: yes
--- a/src/test/test-capability.c
+++ b/src/test/test-capability.c
@@ -254,6 +254,13 @@
 
         ASSERT_OK(capability_get_ambient(&c));
 
+        r = prctl(PR_CAPBSET_READ, CAP_MKNOD);
+        if (r <= 0)
+                return (void) log_tests_skipped("Lacking CAP_MKNOD, skipping getambient test.");
+        r = prctl(PR_CAPBSET_READ, CAP_LINUX_IMMUTABLE);
+        if (r <= 0)
+                return (void) log_tests_skipped("Lacking CAP_LINUX_IMMUTABLE, skipping getambient test.");
+
         r = safe_fork("(getambient)", FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGTERM|FORK_WAIT|FORK_LOG, NULL);
         ASSERT_OK(r);
 
